aten/src/ATen/miopen/Descriptors.cpp

```diff
@@ -0,0 +1,120 @@
+#include "Descriptors.h"
+#include <ATen/ATen.h>
+
+namespace at { namespace native {
```

Comment:
So, I remember a week ago, you told me that `miopen` was API compatible with `cudnn`, and so we were going to regex our way to victory. In this PR, I see a copy of code for `miopen`, which is basically copied straight from the existing cudnn bindings. What constraints changed that lead us here?

Reply:
If we're going to actually do the "fork the code and maintain it as actual source", we had better also fix up all the references to CuDNN in the comments...